packages/experience-builder-sdk/src/hooks/useFetchByBase.ts

```diff
@@ -0,0 +1,30 @@
+import { useEffect, useState } from 'react';
+import { EntityStore } from '@contentful/experience-builder-core';
+import type { Experience } from '@contentful/experience-builder-core/types';
+
+export const useFetchByBase = (fetchMethod: () => Promise<Experience<EntityStore> | undefined>) => {
```

Comment:
I really like that this new helper hook enriches async methods with a loading & error state.